src/libs/actions/IOU.ts

```diff
@@ -1226,15 +1226,15 @@ function getUpdateMoneyRequestParams(
         });
     }
 
-    if (policy?.id && updatedTransaction) {
+    if (PolicyUtils.isPaidGroupPolicy(policy) && updatedTransaction) {
         const currentTransactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`] ?? [];
         optimisticData.push(
             ViolationsUtils.getViolationsOnyxData(
                 updatedTransaction,
                 currentTransactionViolations,
-                !!policy.requiresTag,
+                !!policy?.requiresTag,
```

Comment:
Do we need the `?`s here and below? if `PolicyUtils.isPaidGroupPolicy(policy)` returns true then policy is an object right?

Same question for the ViolationsUtils.getViolationsOnyxData call below on line 2670

Reply:
According to TypeScript we would (and generally TS linters will error if you're optionally chaining unnecessarily). I think if we add a check that policy exists in the if statement above, TS will narrow the type in that scope and know it exists, but from just the `isPaidGroupPolicy` call, TS can't do that narrowing. (It could if `isPaidGroupPolicy` was a type guard that narrowed a nullable general policy type to a paid policy type, or just a nullable policy to a non-null policy, but it's just a predicate).

Reply:
Looks like in this call, the type of `policy` is `OnyxEntry<OnyxTypes.Policy>`, which the `OnyxEntry` wrapper types the value as `TOnyxValue | null` and so the optional chaining is to satisfy the possibility of it resolving as `null`.

Further down in the other `ViolationsUtils.getViolationsOnyxData` call the `policy` is typed as just `OnyxTypes.Policy` so there is no need for the chaining down there.

Reply:
You mean like this? If so, that sounds like a good improvement to me
```
if (policy && PolicyUtils.isPaidGroupPolicy(policy) && updatedTransaction) {
```